### ISSUE TITLE
prometheus connection reset by peer

### ISSUE THREAD

**Issue by LawyZheng** [prometheus/prometheus#10853]
### What did you do?

Similar to #7785, I scraped an exporter behind F5, using HTTPS.

My config is like this
``` yaml
  - job_name: "myjob"
    scheme: "https"
    metrics_path: "/prometheus/metrics"
    basic_auth:
      username: "myusernmae"
      password: "mypassword"
    static_configs:
      - targets: ["myhost.domain.com"]
        labels:
          instance: "myinstance"
```

I did some research about this and located where the problem lies.
It seems `https://myhost.domain.com/prometheus/metrics` is fine to scrape, while `https://myhost.domain.com:443/prometheus/metrics` is not.
Just adding the port number of the domain, will cause the connection to fail.

I don't know why, but It's definitely relative to F5 because I tried with other HTTPS domains which are all fine with or without the port number.
But I still got some questions about the source code, which might be something to do with this "bug" I guess.

https://github.com/prometheus/prometheus/blob/a84c472745123e7c4e319a31dec7e2d14442c2bc/scrape/scrape.go#L759-L771

These codes show that a new request is newed with the URL built by `s.URL().String()`.
`s.URL()` returns `*url.URL` by these code.

https://github.com/prometheus/prometheus/blob/a84c472745123e7c4e319a31dec7e2d14442c2bc/scrape/target.go#L221-L226

In this case, the URL host is supposed to be the same as the value of the "\_\_address\_\_" label, 
so I went to check the label value, but I got the host without the port number in the label.
<img width="426" alt="image" src="https://user-images.githubusercontent.com/26927551/172997653-64a83690-4d4d-49b1-a290-9897a6cf4b82.png">

Maybe I misunderstand the source codes.
I just want to know the reason and what I should do to adjust my config file to fix this problem.

### What did you expect to see?

Successfully Scrape Metrics.

### What did you see instead? Under which circumstances?

I got `read: connection reset by peer` error.
<img width="1245" alt="image" src="https://user-images.githubusercontent.com/26927551/172994786-38c618ce-7267-43e2-b24d-37b4d6d7acb3.png">

Of Course, it works out if I use `curl` or `wget` on my machine/container.

### System information

Kubernetes/Docker/MacOS M1/Windows

### Prometheus version

```text
v2.36.0
```

**Comment by LeviHarrison** [comment]
Hi there! Prometheus does add the default HTTPS port to requests, which may be causing your problem. I while ago I wrote a PR (#9523) which adds an option to disable this behavior, and although it hasn't been merged yet, I think it should work. Maybe that will fix things. If you'd like to try it out, you can grab a build from the [project's CI system](https://app.circleci.com/pipelines/github/prometheus/prometheus/17277/workflows/0d800a8f-9e3b-482f-ad52-0f226bc8c4b0/jobs/83403/artifacts) or clone the branch and build it yourself using the "build from source" instructions found in the READme. Pass the flag `--enable-feature=no-default-scrape-port` to disable adding the port.

**Comment by LawyZheng** [comment]
Thanks a lot. I think it will help me a lot. I will try it as soon as possible, when the PR merged into main branch and releasing the new version.

**Comment by roidelapluie** [comment]
`--enable-feature=no-default-scrape-port` has been merged and released in Prometheus 2.38.
